tests/CsvReadTest.php

```diff
@@ -398,13 +283,174 @@ public function testSkipsOverflow()
+        new CsvReader(__DIR__ . '/data/test-input.csv', ",", $enclosure);
+    }
+
+    public function invalidEnclosureProvider()
+    {
+        return [
+            ['aaaa', 'Enclosure must be a single character. "aaaa" received'],
+            ['ob g', 'Enclosure must be a single character. "ob g" received'],
+        ];
+    }
+
+    /**
+     * @dataProvider invalidSkipLinesProvider
+     * @param mixed $skipLines
+     * @param string $message
+     */
+    public function testInvalidSkipLines($skipLines, $message)
+    {
+        self::expectException(Exception::class);
+        self::expectExceptionMessage($message);
+        new CsvReader(
+            'dummy',
+            CsvReader::DEFAULT_DELIMITER,
+            CsvReader::DEFAULT_ENCLOSURE,
+            CsvReader::DEFAULT_ENCLOSURE,
+            $skipLines
+        );
+    }
+
+    public function invalidSkipLinesProvider()
+    {
+        return [
+            ['invalid', 'Number of lines to skip must be a positive integer. "invalid" received.'],
+            [-123, 'Number of lines to skip must be a positive integer. "-123" received.']
+        ];
+    }
+
+    public function testInvalidNewLines()
+    {
+        self::expectException(Exception::class);
+        self::expectExceptionMessage('Invalid line break. Please use unix \n or win \r\n line breaks.');
+        $csvFile = new CsvReader(__DIR__ . DIRECTORY_SEPARATOR . 'data/binary');
+    }
+
+
+    public function testValidWithoutRewind()
+    {
+        $fileName = __DIR__ . '/data/simple.csv';
+
+        $csvFile = new CsvReader($fileName);
+        self::assertTrue($csvFile->valid());
+    }
+
+    public function testHeaderNoReset()
+    {
+        $fileName = __DIR__ . '/data/simple.csv';
+
+        $csvFile = new CsvReader($fileName);
+        $csvFile->rewind();
+        self::assertEquals(['id', 'isImported'], $csvFile->current());
+        $csvFile->next();
+        self::assertEquals(['15', '0'], $csvFile->current());
+        self::assertEquals(['id', 'isImported'], $csvFile->getHeader());
+        self::assertEquals(['15', '0'], $csvFile->current());
+    }
+
+    public function testLineBreakWithoutRewind()
+    {
+        $fileName = __DIR__ . '/data/simple.csv';
+
+        $csvFile = new CsvReader($fileName);
+        self::assertEquals("\n", $csvFile->getLineBreak());
+    }
+
+    public function testWriteReadInTheMiddle()
+    {
+        $fileName = sys_get_temp_dir() . DIRECTORY_SEPARATOR . uniqid('csv-test');
+        $writer = new CsvWriter($fileName);
+        $reader = new CsvReader($fileName);
+        self::assertEquals([], $reader->getHeader());
+        $rows = [
+            [
+                'col1', 'col2',
+            ],
+            [
+                '1', 'first',
+            ],
+            [
+                '2', 'second',
+            ],
+        ];
+
+        $writer->writeRow($rows[0]);
+        $reader->next();
+        self::assertEquals(false, $reader->current());
```

Comment:
jakto? Neměl by tam být už ten zapsaný řádek?

Reply:
Bylo by fajn tam dát komentář nebo message k tomu assertu

Reply:
nemel, protoze ten reader je nakonci, musi se mu tam dat rewind
je to counterintuitive a nereprezentuje to zadnej usecase, ale je to dokumentace toho jak se to chova pokud udelas takovyhle zverstvo :) Nicmene porad je to lepsi nez stavajici verze, ktera je schopna si prepsat existujici radky (protoze ma stejny pointer pro read i write)